lib/assert.js

```diff
@@ -69,6 +69,22 @@ x.notSame = function (val, expected, msg) {
 };
 
 x.throws = function (fn, err, msg) {
+	if (fn && fn.then) {
+		err = err || Error;
+
+		return fn
+			.catch(err, function (err) {
+				return err;
+			})
+			.then(function (value) {
+				if (value instanceof err) {
+					return;
+				}
+
+				test(false, create(undefined, undefined, undefined, msg || 'Missing expected exception', x.throws));
```

Comment:
Why is `actual` and `expected` undefined? Shouldn't we follow the sync method: https://github.com/sindresorhus/core-assert/blob/494c1eaea60090d4d4a187f03de9c2f4051b6f9b/index.js#L321

As for `msg`, see: https://github.com/sindresorhus/core-assert/blob/494c1eaea60090d4d4a187f03de9c2f4051b6f9b/index.js#L317-L318

Maybe it would be easier to just use `assert.throws(fn, err, msg);` here.